### PR TITLE
METAL-2746 check and skip if gsql instance is not ready for a new update

### PR DESCRIPTION
**Pull request by dabde**
Avoid requesting multiple update request against gcloud, if the instance is still not ready.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,53 +2,48 @@
...
+update: build deploy ## build db-operator image again and delete running pod
 
-test:
-	@docker-compose up -d
-	@docker-compose restart sqladmin
-	@sleep 2
-	@go test -count=1 -tags tests ./... -v -cover
-	@docker-compose down
+test: ## spin up mysql, postgres containers and run go unit test
+	tearDown() {
```
Comment (dabde) [inline Makefile:39]: avoid docker-compose still run if the target failed. Think mainly an issue on local side.

File: Makefile [hunk 2]
```diff
@@ -2,53 +2,48 @@
...
-	@echo "make minidown: stop minikube"
-	@echo "make minidelete: delete minikube"
-	@echo "make minidashboard: open minikube dashboard"
-	@echo "make build: build db-operator docker image"
-	@echo "make helm: install helm if not exist and install local chart using helm upgrade --install command"
-	@echo "make setup: build db-operator image, install helm"
-	@echo "make update: build db-operator image again and delete running pod"
-	@echo "make addexamples: kubectl create -f examples/"
-	@echo "make test: spin up mysql, postgres containers and run go unit test"
-	@echo "make microsetup: install microk8s locally and deploy db-operator (only for linux and mac)"
```
Comment (dabde) [inline Makefile:18]: was not existing. so update the whole makefile.

File: pkg/utils/dbinstance/service.go [hunk 3]
```diff
@@ -49,6 +49,15 @@ func Update(ins DbInstance) (map[string]string, error) {
 		return nil, ErrNotExists
 	}
 
+	state, err := ins.state()
+	if err != nil {
+		return nil, err
+	}
+
+	if (state != "RUNNABLE") && (state != "NOT_SUPPORTED") {
```
Comment (dabde) [inline pkg/utils/dbinstance/service.go:57]: don't know if I like this construct. open for other solutions.

Comment (dpeckett) [reply pkg/utils/dbinstance/service.go:57]: Just remove the extra parentheses and we are good.

File: pkg/utils/dbinstance/errors.go [hunk 4]
```diff
@@ -23,4 +23,6 @@ var (
 	ErrAlreadyExists = errors.New("instance already exists")
 	// ErrNotExists is thrown when db instance does not exists
 	ErrNotExists = errors.New("instance does not exists")
+	// ErrInstanceNotReady is throw is gsql instance is still not marked as Ready
```
Comment (dpeckett) [inline pkg/utils/dbinstance/errors.go:26]: English grammer, "is thrown when"

File: Makefile (hunk 1 above)
Comment (dpeckett) [reply Makefile:38]: Does this work? Definitely an interesting construct to see in a makefile.

Comment (dabde) [reply Makefile:38]: yes, tested before fixed my unit test error.

File: pkg/utils/dbinstance/service.go (hunk 3 above)
Comment (dabde) [reply pkg/utils/dbinstance/service.go]: fixed.

File: pkg/utils/dbinstance/errors.go (hunk 4 above)
Comment (dabde) [reply pkg/utils/dbinstance/errors.go]: fixed

**Review by dpeckett — APPROVED** [review]

**Pull request merged**